### DISCUSSION TITLE
[Linq2DB CLI] Generating mappings with aliased names?

### DISCUSSION THREAD

**Discussion by Metadorius** [linq2db/linq2db#3737] [Q&A] [answered]
Hello!

I've been working on mappings for a big obfuscated DB (changing the column and table names is out of question because it's used by a lot of legacy applications) and wondered if there's any way to supply something like custom rename rules so that the CLI can process the names found in the database into something more human-readable. Ideally I would write something like a list of match-replace regexes for column names, table names, etc. in the JSON that the CLI tool is using. Is something like this currently possible to achieve or would be easy to implement?

**Comment by MaceWindu** [comment, marked as answer]
Only option for now is to use PreprocessEntity interceptor https://linq2db.github.io/articles/CLI.html#data-model-interceptors

**Reply by MaceWindu** [reply]
I've created feature request #3739
